sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DDLParserSuite.scala

```diff
@@ -784,6 +784,16 @@ class DDLParserSuite extends AnalysisTest {
         Some(EqualTo(UnresolvedAttribute("t.a"), Literal(2)))))
   }
 
+  test("delete from table: cte") {
+    parseCompare("WITH c(a) AS (SELECT 2) DELETE FROM testcat.ns1.ns2.tbl AS t WHERE t.a = c.a",
+      With(DeleteFromTable(
```

Comment:
It's pretty straightforward that the parser will wrap the plan with a `With`, I think the part we need to verify is the analyzer. Can we move tests to `PlanResolutionSuite` to test analyzing behavior?

Reply:
Yeah, you're right, we should do that. Will update soon.